### PR TITLE
API-5168: added connection to xml services to allow retrieving combined apis (rest and xml)

### PR DESCRIPTION
**Pull request by mattclark-zerogravit**

### CODE REVIEW

File: app/uk/gov/hmrc/apiplatformmicroservice/combinedapis/models/JsonFormatters.scala [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.apiplatformmicroservice.combinedapis.models
+
+import play.api.libs.json.{Format, Json}
+import uk.gov.hmrc.apiplatformmicroservice.apidefinition.models.ApiCategory
+
+trait BasicCombinedApiJsonFormatters{
+  implicit val apiCategoryFormat: Format[ApiCategory] = Json.format[ApiCategory]
```
Comment (anjumabbas5) [inline app/uk/gov/hmrc/apiplatformmicroservice/combinedapis/models/JsonFormatters.scala:23]: This formatter also exists in app/uk/gov/hmrc/apiplatformmicroservice/xmlapis/models/JsonFormatters.scala

File: app/uk/gov/hmrc/apiplatformmicroservice/xmlapis/models/JsonFormatters.scala [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.apiplatformmicroservice.xmlapis.models
+
+import play.api.libs.json.{Format, Json}
+import uk.gov.hmrc.apiplatformmicroservice.apidefinition.models.ApiCategory
+
+trait BasicXmlApisJsonFormatters {
+  implicit val formatApiCategory: Format[ApiCategory] = Json.valueFormat[ApiCategory]
```
Comment (anjumabbas5) [inline app/uk/gov/hmrc/apiplatformmicroservice/xmlapis/models/JsonFormatters.scala:23]: This formatter also exists in app/uk/gov/hmrc/apiplatformmicroservice/combinedapis/models/JsonFormatters.scala

File: app/uk/gov/hmrc/apiplatformmicroservice/combinedapis/models/JsonFormatters.scala (hunk 1 above)
Comment (mattclark-zerogravit) [reply app/uk/gov/hmrc/apiplatformmicroservice/combinedapis/models/JsonFormatters.scala:23]: Both xmlApi and CombinedApi use apiCategory so both need the formatter.

File: app/uk/gov/hmrc/apiplatformmicroservice/xmlapis/models/JsonFormatters.scala (hunk 2 above)
Comment (mattclark-zerogravit) [reply app/uk/gov/hmrc/apiplatformmicroservice/xmlapis/models/JsonFormatters.scala:23]: Both xmlApi and CombinedApi use apiCategory so both need the formatter.

**Review by anjumabbas5 — APPROVED** [review]

File: app/uk/gov/hmrc/apiplatformmicroservice/combinedapis/services/CombinedApisService.scala [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+                                   (implicit ec: ExecutionContext) {
+
+
+
+  def fetchCombinedApisForDeveloperId(developerId: Option[DeveloperIdentifier])
+                                     (implicit hc: HeaderCarrier): Future[List[CombinedApi]] =
+    for{
+      restApis <- apiDefinitionsForCollaboratorFetcher.fetch(developerId)
+      xmlApis <- xmlApisConnector.fetchAllXmlApis()
+    } yield restApis.map(fromApiDefinition) ++ xmlApis.map(fromXmlApi)
```
Comment (AndySpaven) [inline app/uk/gov/hmrc/apiplatformmicroservice/combinedapis/services/CombinedApisService.scala:40]: Should we kick these off in parallel given they could be long-ish lists that take a while ?

File: app/uk/gov/hmrc/apiplatformmicroservice/xmlapis/connectors/XmlApisConnector.scala [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+
+  private lazy val serviceBaseUrl: String = appConfig.serviceBaseUrl
+
+  def fetchAllXmlApis()(implicit hc: HeaderCarrier): Future[Seq[XmlApi]] = {
+
+    logger.info(s"${this.getClass.getSimpleName} - fetchAllXmlApis")
+
+    val r = httpClient.GET[Seq[XmlApi]](s"$serviceBaseUrl/api-platform-xml-services/xml/apis")
+
+    r.recover {
```
Comment (AndySpaven) [inline app/uk/gov/hmrc/apiplatformmicroservice/xmlapis/connectors/XmlApisConnector.scala:43]: We should probably wrap this up in a function (well partial function anyhow) and reuse.

Comment (mattclark-zerogravit) [reply app/uk/gov/hmrc/apiplatformmicroservice/xmlapis/connectors/XmlApisConnector.scala]: I've made this change now. I think I've changed this how you suggested.

File: app/uk/gov/hmrc/apiplatformmicroservice/combinedapis/services/CombinedApisService.scala (hunk 3 above)
Comment (mattclark-zerogravit) [reply app/uk/gov/hmrc/apiplatformmicroservice/combinedapis/services/CombinedApisService.scala:40]: The 2nd list is static and not that long. See api-platform-xml-services and look for xml_apis.json.

**Pull request merged**